x-pack/test/api_integration/apis/uptime/rest/telemetry_collectors.ts

```diff
@@ -119,6 +122,8 @@ export default function ({ getService }: FtrProviderContext) {
     });
 
     it('should receive expected results after calling overview logging', async () => {
+      // wait few seconds to make sure data is refreshed, just to avoid flakiness
+      await delay(2000);
       // call overview page
       const { body: result } = await supertest
```

Comment:
I think it might make sense to try wrapping this `supertest` call in a `retry` block instead of hardcoding delay values. In the past when I have attempted to implement similar solutions I have usually gotten pushback because it goes against the typical pattern used when trying to keep these tests maintainable. cc @spalger

Reply:
My concern isn't about consistency, it's about the fact that the delay is only necessary if there's something we should be waiting for but we're not waiting for it.

What should these tests be waiting for? Can we add a wait for that thing? If we can't, then we can try retrying.

Delay's seem to work today, but they age out as performance characteristics change and become very difficult to maintain over time.

Reply:
i agree, will do that